torch_xla/experimental/distributed_checkpoint.py

```diff
@@ -92,7 +92,9 @@ def set_up_planner(self, state_dict: STATE_DICT_TYPE,
         for k, v in state_dict.items()
         if _is_sharded_tensor(v) or isinstance(v, _CpuShards)
     }
-    unsharded = dict(state_dict.items() - self.sharded_state_dict.items())
+    unsharded = {
```

Comment:
Curious on why you made this change?

Reply:
This is to support `_CpuShard`. The old approach tries to hash the values of the dict as well, since `.items()` is `(key, value)` pairs. When we're checkpointing on CPU, the `_CpuShards` contains `List[XLAShard]` which isn't hashable.

Reply:
Hmm, but  unsharded  is still a KV pair?

Reply:
The issue is with `state_dict.items() - self.sharded_state_dict.items()` - the `-` operator between two `dict_items` is a set difference, which will hash the entire `(k, v)` tuple. Using dict comprehension, only `k` is hashed.

Reply:
That's interesting. Thanks for the python lecture!